packages/api/src/routes/user.ts

```diff
@@ -175,8 +176,17 @@ async function revokeUserProviderAccess(
     const cxId = connectedUser.dataValues.cxId;
     if (token) await Constants.PROVIDER_OAUTH1_MAP[providerOAuth1.data].deregister([token], cxId);
   } else if (providerNoAuth.success) {
-    const noAuthProvider = new Constants.noAuthProviders[providerNoAuth.data]();
-    await noAuthProvider.revokeProviderAccess(connectedUser);
+    if (providerNoAuth.data === PROVIDER_TENOVI) {
+      const tenovi = new Constants.noAuthProviders[PROVIDER_TENOVI]();
+      const rawParams: RawParams = {
+        query: { ...(req.query as RawParams["query"]) },
+        headers: { ...(req.headers as RawParams["headers"]) },
+      };
+      await tenovi.revokeProviderAccess(connectedUser, rawParams);
+    } else {
+      const noAuthProvider = new Constants.noAuthProviders[providerNoAuth.data]();
+      await noAuthProvider.revokeProviderAccess(connectedUser);
+    }
```

Comment:
We could pass `rawParams` to `revokeProviderAccess()`, we just need to make a `NoAuth` interface like we have for `OAuth1` and `OAuth2`, with a function `revokeProviderAccess(connectedUser: ConnectedUser, rawParams?: RawParams): Promise<void>`, and make `Apple` and `Tenovi` to implement that interface (as we should've done since the beginning 😄).

Then we can remove the Tenovi specific code here (that's why we came up with the `noAuth` approach in the first place)

Reply:
Yea, I actually did think that was weird, but wasn't sure how to abstract it out from this code. Thanks for the heads up! 🙏